util/benchmark/ec/p256_awslc_ossl/src/benchmark_ecdsa.c

```diff
@@ -0,0 +1,151 @@
+    key = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
+
+    if (!key ||
+        /* Generate key */
+        !EC_KEY_generate_key(key) ||
+        /* Check key size */
+        ECDSA_SIGNATURE_BYTE_SIZE < ECDSA_size(key)
+        )
+    {
+        ecdsa_checks = 0;
+        BIO_printf(bio_err, "ECDSA EC key generation failure.\n");
+        ERR_print_errors(bio_err);
+    }
+
+    if (1 == ecdsa_checks)
+    {
+        /* Pre-fill digest buffer */
+        OPENSSL_memset(digest, 42, sizeof(digest));
+
+        /* Test ECDSA signing and verifying*/
+        if (!ECDSA_sign(0, digest, sizeof(digest), signature, &sig_len, key) ||
+            !ECDSA_verify(0, digest, sizeof(digest), signature, sig_len, key)
+            )
+        {
+            ecdsa_checks = 0;
+            BIO_printf(bio_err, "ECDSA signing and verifying failure.\n");
+            ERR_print_errors(bio_err);
+        }
+
+    }
+
+    if (1 == ecdsa_checks)
+    {
+        /* Benchmark ECDSA signing */
+        start = time_now();
+#if defined(PID_CPU_TICKS)
+        cpu_ticks_start = cpu_now(fpstat, &flags);
+#endif
+
+        for (int i = 0; i < num_itr_sign; i++)
+        {
+            ECDSA_sign(0, digest, sizeof(digest), signature, &sig_len, key);
+        }
+
+        now = time_now();
+#if defined(PID_CPU_TICKS)
+        cpu_ticks_end = cpu_now(fpstat, &flags);
+#endif
+        us = now - start;
+        BIO_printf(bio_out, "ECDSA P-256 sign: %u operations in %luus (%.1f ops/sec)\n",
+                   num_itr_sign, (long unsigned)us,
+                   ((double)num_itr_sign/us) * 1000000);
+#if defined(PID_CPU_TICKS)
+        cpu_ticks = cpu_ticks_end - cpu_ticks_start;
+        BIO_printf(bio_out, "            in %ld cpu ticks\n",
+                   cpu_ticks);
+
+        // Close /proc/<pid>/stat
+        close_fpstat(fpstat);
+#endif
+
+        /* Benchmark ECDSA verification */
+#if defined(PID_CPU_TICKS)
+        // Open /proc/<pid>/stat
+        fpstat = open_fpstat();
+#endif
+
+        start = time_now();
+#if defined(PID_CPU_TICKS)
+        cpu_ticks_start = cpu_now(fpstat, &flags);
+#endif
+        for (int i = 0; i < num_itr; i++)
+        {
+            ECDSA_verify(0, digest, sizeof(digest), signature, sig_len, key);
+        }
+
+        now = time_now();
+#if defined(PID_CPU_TICKS)
+        cpu_ticks_end = cpu_now(fpstat, &flags);
+#endif
+        us = now - start;
+        BIO_printf(bio_out, "ECDSA P-256 verify: %u operations in %luus (%.1f ops/sec)\n",
+                   num_itr, (long unsigned)us,
+                   ((double)num_itr/us) * 1000000);
+#if defined(PID_CPU_TICKS)
+        cpu_ticks = cpu_ticks_end - cpu_ticks_start;
+        BIO_printf(bio_out, "            in %ld cpu ticks\n",
+                   cpu_ticks);
+        // Close /proc/<pid>/stat
+        close_fpstat(fpstat);
+#endif
+    }
+
+    if (1 == ecdsa_checks)
+    {
+        BIO_printf(bio_out, "SUCCESS: ECDSA Test.\n");
+    }
+
+    EC_KEY_free(key);
+
+	return;
```

Comment:
void return type.

Reply:
I think he wants to remove the "return" since the function is void